### PR TITLE
Add DataprocDeleteClusterOperatorAsync

### PR DESCRIPTION
**Pull request by pankajastro**
Issue: https://github.com/astronomer/astronomer-providers/issues/424

- Add Example DAG
- Add hook/trigger/operator
- Add Unit Test

### CODE REVIEW

**Comment by rajaths010494** [comment]
Code coverage is failing

**Review by bharanidharan14 — COMMENTED** [review]
Fix Test Coverage

File: astronomer/providers/google/cloud/triggers/dataproc.py [hunk 1]
```diff
@@ -87,6 +88,78 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
...
+                    metadata=self.metadata,
+                )
+                self.log.info(
+                    "Cluster status is %s. Sleeping for %s seconds.",
+                    cluster.status.state,
+                    self.polling_interval,
+                )
+                await asyncio.sleep(self.polling_interval)
+            except NotFound:
+                yield TriggerEvent({"status": "success", "message": ""})
```
Comment (phanikumv) [inline astronomer/providers/google/cloud/triggers/dataproc.py:157]: why arent we sending message something like "cluster with id <cluster id> deleted successfully" here ?

Comment (pankajastro) [reply astronomer/providers/google/cloud/triggers/dataproc.py:157]: sync version just log `Cluster deleted` in case of success

**Review by phanikumv — APPROVED** [review]

**Review by rajaths010494 — APPROVED** [review]

**Pull request merged**